tamr_toolbox/notifications/common.py

```diff
@@ -1,12 +1,123 @@
+                instead of a pre-defined notification type
+        """
+        raise NotImplementedError(
+            "Do not call `_BaseNotifier` directly. " "Use e.g. `EmailNotifier` instead."
+        )
+
+    def monitor_job(
+        self,
+        tamr: Client,
+        operation: Union[int, str, Operation],
+        notify_states: Optional[List[OperationState]] = None,
+        poll_interval: Optional[float] = 1,
+        timeout: Optional[float] = None,
+    ) -> None:
+        """Monitors a Tamr Operation and sends a message when the job status is updated
+
+        Args:
+            tamr: A Tamr client
+            operation: A job ID or a Tamr operation
+            notify_states : States for which notifications should be sent, use None for all states
+            poll_interval: Time interval (in seconds) between subsequent polls
+            timeout: Time (in seconds) to wait before timing out
+
+        Returns:
+            A list of messages with their response codes
+        """
+        if not notify_states:
+            notify_states = [
+                OperationState.SUCCEEDED,
+                OperationState.FAILED,
+                OperationState.CANCELED,
+                OperationState.PENDING,
+                OperationState.RUNNING,
+            ]
+
+        if isinstance(operation, Operation):
+            op = operation
+        else:
+            op = from_resource_id(tamr=tamr, job_id=operation)
+
+        # Send message for initial state
+        status = OperationState[op.state]
+        if status in notify_states:
+            self.send_message(
+                message=get_details(op), title=f"Job {operation.resource_id}: {status}"
+            )
+
+        while status not in [
+            OperationState.SUCCEEDED,
+            OperationState.FAILED,
+            OperationState.CANCELED,
+        ]:
+
+            # Send message on state change
+            try:
+                op = monitor(
+                    operation=op, poll_interval_seconds=poll_interval, timeout_seconds=timeout
+                )
+
+                status = OperationState[op.state]
+                self.send_message(
+                    message=get_details(op), title=f"Job {operation.resource_id}: {status}"
+                )
+            except TimeoutError:
+                timeout_message = (
+                    f"The job {op.resource_id}: {op.description} took longer "
+                    f"than {timeout} seconds to resolve."
+                )
+                self.send_message(message=timeout_message, title=f"Job {op.resource_id}: Timeout")
+
+    def _parse_recipients(self, tamr_user: Optional[str] = None) -> List[str]:
+        """Look up Tamr users & parse recipients into an iterable list"""
+        if type(self.recipients) in [str, list]:
+            if tamr_user:
+                raise ValueError(
+                    "If specifying a Tamr user, the recipients class input"
```

Comment:
could you use this in one example please?

Reply:
Done.

Slack notification example is setup as a string recipient.
Teams notification example is setup as a list of strings recipients
Email notification example is setup as a dict of tamr users:emails recipients